test/integration/vorpal-engine_test.go

```diff
@@ -34,26 +39,84 @@ func TestScanVorpal_SentFileWithoutExtension_FailCommandWithError(t *testing.T)
 	assert.ErrorContains(t, err, errorConstants.FileExtensionIsRequired)
 }
 
-func TestExecuteVorpalScan_VorpalLatestVersionSetTrue_SuccessfullyReturnMockData(t *testing.T) {
+func TestExecuteVorpalScan_VorpalLatestVersionSetTrue_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "",
+		flag(commonParams.VorpalLatestVersion),
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", true)
-	expectedMockResult := commands.ReturnSuccessfulResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, "Sending empty source file should not fail")
 }
 
-func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_SuccessfullyReturnMockData(t *testing.T) {
+func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", false)
-	expectedMockResult := commands.ReturnFailureResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
 }
 
 func TestExecuteVorpalScan_CorrectFlagsSent_SuccessfullyReturnMockData(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
+}
+
+func TestExecuteVorpalScan_InitializeAndRun_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "",
+		flag(commonParams.VorpalLatestVersion),
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	vorpalWrapper := grpcs.NewVorpalGrpcWrapper(viper.GetInt(commonParams.VorpalPortKey))
+	if healthCheckErr := vorpalWrapper.HealthCheck(); healthCheckErr == nil {
+		_ = vorpalWrapper.ShutDown()
+	} else {
+		t.Failed()
```

Comment:
if this test will run first - the test will fail here

Reply:
Yes I fixed it already